streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1086,6 +1088,34 @@ private boolean nodeGroupContainsGlobalSourceNode(final Set<String> allNodesOfGr
         return false;
     }
 
+    private static class NodeComparator implements Comparator<org.apache.kafka.streams.TopologyDescription.Node>, Serializable {
+
+        @Override
+        public int compare(final org.apache.kafka.streams.TopologyDescription.Node node1,
+                           final org.apache.kafka.streams.TopologyDescription.Node node2) {
+            final int size1 = ((AbstractNode) node1).size;
+            final int size2 = ((AbstractNode) node2).size;
+
+            // it is possible that two nodes have the same sub-tree size (think two nodes connected via state stores)
+            // in this case default to processor name string
+            if (size1 != size2)
+                return size2 - size1;
+            else
+                return node1.name().compareTo(node2.name());
+
+        }
+    }
+
+    private final static NodeComparator NODE_COMPARATOR = new NodeComparator();
+
+    private static void updateSize(final AbstractNode node, final int delta) {
```

Comment:
nit: parameter/line formatting

Reply:
I thought we will only need to break lines if the parameters cannot fit in a single line.

Reply:
I think for calling methods single line is fine. But for defining method, we should always go with one parameter per line.

Reply:
Do we have that rule actually? We have lots of method definitions where multiple parameters sits in the single line. I think we would only have multiple lines if all the parameters do not fit in a single line.

Reply:
No we don't have that rule. Personally i think it is fine as long as it fits on a single line, i.e., less than 100 characters.

Reply:
Current code doesn't follow this. It's of course a matter of taste (as always). I think it makes sense as it makes diffs easier to read on Github if a parameter gets added/removed. Thus, going forward, I would prefer this formatting (and we should reformat "old" code incrementally -- similar to `final`).

Btw: I added this rule to the new coding guidelines on the Kafka webpage. If we don't want the rule, we should remove it. Also, the guideline says 120 chars at max length. I am fine with 100, too, if there is a strong opinion on this.